client/vendors/ngecharts.js

```diff
@@ -89,6 +99,11 @@
                 createChart(newVal);
             });
 
+            scope.$watch('dataset', function (newVal, oldVal) {
+                if (angular.equals(newVal, oldVal)) return;
```

Comment:
please use the third argument of $warch instead of custom equals comparison
see https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$watch